### PR TITLE
klevernotes: init at 1.0.0

### LINKED ISSUE #342618: Package request: KleverNotes
**Project description**

A note-taking and management application using markdown.

**Metadata**

* homepage URL: https://apps.kde.org/klevernotes/
* source URL: https://invent.kde.org/office/klevernotes
* license: gpl3+
* platforms: linux

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

### PR DESCRIPTION
**Pull request by Iorvethe**
## Description of changes

This add KleverNotes, a new note-taking and management application using markdown. The derivation is mostly copied from [Marknote’s one](https://github.com/NixOS/nixpkgs/blob/master/pkgs/kde/misc/marknote/default.nix), as they are very similar. Let me know if there is anything more I can do.

Homepage: https://apps.kde.org/klevernotes/

Closes #342618 

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by myclevorname** [comment]
What about putting the package in `by-name/kl`? That is recommended with new packages.

**Comment by K900** [comment]
`mkKdeDerivation` isn't accessible there, intentionally.

File: pkgs/kde/misc/klevernotes/default.nix [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+  pname = "klevernotes";
+  version = "1.0.0";
+
+  src = fetchurl {
+    url = "mirror://kde/stable/klevernotes/${version}/klevernotes-${version}.tar.xz";
+    hash = "sha256-9nXOaa2/DyJO3nXGbnvPaYjXFrVoAgx52th67l5geOU=";
+  };
+
+  extraBuildInputs = [
+    kio
```
Comment (K900) [inline pkgs/kde/misc/klevernotes/default.nix:19]: kio should already be injected automatically from metadata.

File: pkgs/kde/misc/klevernotes/default.nix [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    qtsvg
+    qtwebengine
+  ];
+
+  meta.license = [ lib.licenses.gpl3Plus ];
```
Comment (K900) [inline pkgs/kde/misc/klevernotes/default.nix:24]: License is definitely not just that: https://invent.kde.org/office/klevernotes/-/tree/master/LICENSES?ref_type=heads

File: pkgs/kde/misc/klevernotes/default.nix (hunk 1 above)
Comment (Iorvethe) [reply pkgs/kde/misc/klevernotes/default.nix]: You’re right, I had an issue with CMake not finding it yesterday, but today’s build without it works fine. Removed it in the last push

File: pkgs/kde/misc/klevernotes/default.nix (hunk 2 above)
Comment (Iorvethe) [reply pkgs/kde/misc/klevernotes/default.nix]: My bad, I didn’t see this part of the repo and took the license from the product homepage only. I added most of them, except the FSFAP which it seems isn’t in `lib.licenses`. Should I create another MR which adds it there and then here, or simply define it locally?

**Pull request merged**